envs/base_env.py

```diff
@@ -62,11 +62,11 @@ def _getSpecificAction(self, action):
 
     """
     primative_idx, x_idx, y_idx, z_idx, rot_idx = map(lambda a: self.action_sequence.find(a), ['p', 'x', 'y', 'z', 'r'])
-    motion_primative = action[primative_idx] if primative_idx != -1 else 0
-    x = action[x_idx]
-    y = action[y_idx]
-    z = action[z_idx] if z_idx != -1 else self._getPrimativeHeight(motion_primative, x, y)
-    rot = action[rot_idx] if rot_idx != -1 else 0
+    motion_primative = action[primative_idx].item() if primative_idx != -1 else 0
```

Comment:
These should never be pytorch variables. The things sent to the envs should be numpy vars only. If we don't enforce this weird things can start happening with the serialization over threads.

Reply:
You are right. My mistake